boa/src/builtins/number/mod.rs

```diff
@@ -159,6 +159,145 @@ pub fn to_precision(this: &mut Value, args: &[Value], _ctx: &mut Interpreter) ->
     unimplemented!("TODO: Implement toPrecision");
 }
 
+const BUF_SIZE: usize = 2200;
+
+// https://golang.org/src/math/nextafter.go
+#[inline]
+fn next_after(x: f64, y: f64) -> f64 {
+    if x.is_nan() || y.is_nan() {
+        f64::NAN
+    } else if (x - y).abs() < f64::EPSILON {
+        x
+    } else if x == 0.0 {
+        f64::from_bits(1).copysign(y)
+    } else if y > x || x > 0.0 {
+        f64::from_bits(x.to_bits() + 1)
+    } else {
+        f64::from_bits(x.to_bits() - 1)
+    }
+}
+
+// https://stackoverflow.com/questions/55690397/where-is-the-frexp-function-for-f32-in-rust
+#[inline]
+fn frexp(s: f64) -> (f64, isize) {
+    if 0.0 == s {
+        (s, 0)
+    } else {
+        let lg = s.abs().log2();
+        let x = (lg - lg.floor() - 1.0).exp2();
+        let exp = lg.floor() + 1.0;
+        (s.signum() * x, exp as isize)
+    }
+}
+
+// https://chromium.googlesource.com/v8/v8/+/refs/heads/master/src/numbers/conversions.cc#1230
+pub fn num_to_string(mut value: f64, radix: u32) -> String {
+    assert!(radix >= 2);
+    assert!(radix <= 36);
+    assert!(value.is_finite());
```

Comment:
I think these should be debug asserts, since we should already be checking before sending it here:
```suggestion
    debug_assert!(radix >= 2, "radix too small");
    debug_assert!(radix <= 36, "radix too big");
    debug_assert!(value.is_finite(), "non-finite number found");
```

Reply:
i left them as assert since these kind of asserts allow llvm to do further optimizations. I dont know if they apply here but for array length checks they eliminate the bound checks from llvm.